RelayDotNet/Relay.cs

```diff
@@ -1946,5 +1949,186 @@ public async Task<string[]> SetUserProfile(IRelayWorkflow relayWorkflow, string
                 }
             );
         }
+
+        string serverHostname = "all-main-pro-ibot.nocell.io";
+        string version = "relay-sdk-dotnet/2.0.0";
+        string auth_hostname = "auth.relaygo.com";
+
+        private async Task<string> UpdateAccessToken(string refreshToken, string clientId) {
+            // Create a uri object with the following url
+            string url = $"https://{auth_hostname}/oauth2/token";
+            var grantUrl = new Uri(@url);
+
+            // Create a payload to be sent with the request
+            var grantPayload = new Dictionary<string, string>() {
+                ["grant_type"] = "refresh_token",
+                ["refresh_token"] = refreshToken,
+                ["client_id"] = clientId
+            };
+
+            // Create an http client that will post the request
+            HttpClient httpClient = new HttpClient();
+
+            // Add headers and the encoded content to the URL
+            httpClient.DefaultRequestHeaders.Accept.Add(
+                new MediaTypeWithQualityHeaderValue("application/json"));
+            httpClient.DefaultRequestHeaders.Add("User-Agent", version);
+            var encodedContent = new FormUrlEncodedContent(grantPayload);
```

Comment:
When the post is made of the grantPayload, I don't think it needs to be encoded in form-url format, but just as serialized JSON. I'm not positive about that, but we can check.

Reply:
Nevermind, this is for the access grant, not for the http trigger itself. I'm good with it as is.